sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/block/authority/AuthorityRuleChecker.java

```diff
@@ -29,39 +29,39 @@
 
     static boolean passCheck(AuthorityRule rule, Context context) {
         String requester = context.getOrigin();
+        boolean contain = false;
 
-        // Empty origin or empty limitApp will pass.
-        if (StringUtil.isEmpty(requester) || StringUtil.isEmpty(rule.getLimitApp())) {
-            return true;
+        {
+            // Empty origin or empty limitApp will pass.
+            if (StringUtil.isEmpty(requester) || StringUtil.isEmpty(rule.getLimitApp())) {
+                return true;
+            }
         }
 
-        // Do exact match with origin name.
-        int pos = rule.getLimitApp().indexOf(requester);
```

Comment:
I think this double check is a performance optimization, I will run a performance test later to see the performance of 'indexOf` and loop.

Reply:
I have test it, The performance of indexOf is nearly 10 times that of loop traversal.  I think should remain the double check because In most cases the requester may not be in the black or white list and will be blocked by the first check

Reply:
Thank you for your correction.I have tested the problem you mentioned, the extra consumption is around 7 times, but it is still very small (600 milliseconds more than 10 million tests).
But through an analysis of the problem you provided, I came up with another bug, which you can see in #1760